src/libstd/net/mod.rs

```diff
@@ -76,6 +76,11 @@ impl Iterator for LookupHost {
     fn next(&mut self) -> Option<io::Result<SocketAddr>> { self.0.next() }
 }
 
+/// Resolve the given address to a hostname.
+pub fn lookup_addr(addr: IpAddr) -> io::Result<String> {
```

Comment:
I think this should take a `SocketAddr` now instead of an `IpAddr` (and better yet a `&SocketAddr`)